app/controllers/v1/tasks_controller.rb

```diff
@@ -59,6 +59,6 @@ def complete
 
   def task_params
     params.require(:task).permit(:supplier_id, :framework_id, :status, :due_on,
-                                 :period_month, :period_year, :description)
+                                 :period_month, :period_year, :description, :supplier_name)
```

Comment:
You shouldn't add `:supplier_name` here. It's the attributes that are whitelisted for use here (https://github.com/dxw/DataSubmissionServiceAPI/pull/152/files#diff-ccc873f4133c19aa7be93616ff932243L31), and supplier name shouldn't be updatable via the Task API.